packages/rich-text/src/component/index.native.js

```diff
@@ -1260,13 +1260,15 @@ export default compose( [
 
 		const settings = getSettings();
 		const baseGlobalStyles = settings?.__experimentalGlobalStylesBaseStyles;
-		const experimentalFeatures =
-			settings?.__experimentalFeatures?.color?.palette;
-		const colorPalette =
-			experimentalFeatures?.user ??
-			experimentalFeatures?.theme ??
-			experimentalFeatures?.default ??
-			settings?.colors;
+		const colorsPalettes = settings?.__experimentalFeatures?.color?.palette;
+		const allColorsPalette = [
+			...( colorsPalettes?.theme || [] ),
+			...( colorsPalettes?.custom || [] ),
+			...( colorsPalettes?.default || [] ),
+		];
```

Comment:
It would be great if in the future, once we transform this component into a functional component, we get the colors palette via `useMobileGlobalStylesColors`.

Reply:
Definitely, this file would be a tricky one to update into a functional component but we will have to do it one day 😅